stdlib/char.gr

```diff
@@ -172,6 +172,42 @@ provide let toString = (char: Char) => {
   WasmI32.toGrain(string): String
 }
 
+/**
+ * Byte encodings
+ *
+ * @since v0.7.0
+ */
+provide enum Encoding {
+  UTF8,
+  UTF16,
+  UTF32,
+}
+
+/**
+ * Returns the byte count of the given character in the given encoding.
+ *
+ * @param encoding: The encoding to use
+ * @param char: The character
+ * @returns The byte count of the character in the given encoding
+ *
+ * @example Char.byteCount(Char.UTF8, 'a') == 1
+ * @example Char.byteCount(Char.UTF8, '🌾') == 4
+ * @example Char.byteCount(Char.UTF16, '©') == 1
+ *
+ * @since v0.7.0
+ */
+@unsafe
+provide let byteCount = (encoding, char: Char) => {
+  let usv = untagChar(char)
+  let utf8ByteCount = usvEncodeLength(usv)
+  let utf8ByteCount = tagSimpleNumber(utf8ByteCount)
+  match (encoding) {
+    UTF32 => 4,
```

Comment:
Can check this at the beginning since it'll always be 4 but nbd

Reply:
I don't think it's worth it as it makes the match non exhaustive and adds an extra branch, `UTF32` probably isn't going to be used very often anyways so optimizing for it doesn't seem worth it.